src/main/java/org/zalando/putittorest/RestClientPostProcessor.java

```diff
@@ -134,8 +149,57 @@ private String registerRest(final String id, final String factoryId, final Strin
         });
     }
 
+    private String registerRestTemplate(final String id, final String factoryId, final String convertersId,
+            @Nullable final String baseUrl) {
+        return registry.register(id, RestTemplate.class, () -> {
+            LOG.debug("Client [{}]: Registering RestTemplate", id);
+
+            final BeanDefinitionBuilder restTemplate = genericBeanDefinition(RestTemplate.class);
+
+            restTemplate.addConstructorArgReference(factoryId);
+
+            final DefaultUriTemplateHandler handler = new DefaultUriTemplateHandler();
+            handler.setBaseUrl(baseUrl);
+            restTemplate.addPropertyValue("uriTemplateHandler", handler);
+
+            final AbstractBeanDefinition converters = BeanDefinitionBuilder.genericBeanDefinition()
+                    .setFactoryMethod("getConverters")
+                    .getBeanDefinition();
+            converters.setFactoryBeanName(convertersId);
+            restTemplate.addPropertyValue("messageConverters", converters);
+
+            return restTemplate;
+        });
+    }
+
+    private String registerAsyncRestTemplate(final String id, final String factoryId, final String convertersId,
+            @Nullable final String baseUrl) {
+        return registry.register(id, AsyncRestTemplate.class, () -> {
+            LOG.debug("Client [{}]: Registering AsyncRestTemplate", id);
+
+            final BeanDefinitionBuilder restTemplate = genericBeanDefinition(AsyncRestTemplate.class);
+
+            restTemplate.addConstructorArgReference(factoryId);
+            restTemplate.addConstructorArgReference(factoryId);
```

Comment:
Would in fact be the very same thing, right. I payed around with it and tried to support two individual factories (sync + async), but I couldn't get it to work. So the current requirement is to have a request factory that implements both. Which is probably fine, since nobody in his right mind will override this :wink: